### DISCUSSION TITLE
Rendering engine with Unity3D

### DISCUSSION THREAD

**Discussion by thatdevin** [carla-simulator/carla#4115] [Ideas]
Hi, this is just an idea just coming up with my mind. Is it possible that Carla can also take Unity into consideration as an rendering engine. I don't know whether you guys already had the discussion before, but it is still reasonable and useful.

**Comment by germanros1987** [comment]
Hi @EVYang1992 we have a lot of experience with both UE4 and Unity3D. In fact, we have built many other projects with Unity3D, e.g. SYNTHIA. 

However, for CARLA we are looking to get something that is as open-source as possible, and where users can have control of all aspects, including changing the entire rendering pipeline if they need to. Unity3D, due to its license does not fit our requirements. In fact, Lumberyard may be a better candidate.

We will continue using UE4 for the time being.

**Reply by xiaoqingnianbudadun** [reply]
may i have a question? i can't find the carla branch of UE4.,  how can i build carla?

**Reply by xiaoqingnianbudadun** [reply]
i find the carla branch of UE4
https://github.com/CarlaUnreal/UnrealEngine
